assets/src/components/account/AccountOverview.tsx

```diff
@@ -18,24 +18,43 @@ import {BASE_URL} from '../../config';
 type Props = {};
 type State = {
   account: any;
+  companyName: string;
   currentUser: any;
   inviteUrl: string;
+  isEditing: boolean;
 };
 
 class AccountOverview extends React.Component<Props, State> {
   input: any = null;
 
-  state: State = {account: null, currentUser: null, inviteUrl: ''};
+  state: State = {
+    account: null,
+    companyName: '',
+    currentUser: null,
+    inviteUrl: '',
+    isEditing: false,
+  };
 
   async componentDidMount() {
-    const [account, currentUser] = await Promise.all([
-      API.fetchAccountInfo(),
-      API.me(),
-    ]);
+    // NB: this fetches the account data and also handles setting this.state.account and this.state.companyName
+    await this.fetchLatestAccountInfo();
 
-    this.setState({account, currentUser});
+    const currentUser = await API.me();
+    this.setState({currentUser});
   }
 
+  fetchLatestAccountInfo = async () => {
+    const account = await API.fetchAccountInfo();
+
+    if (account) {
+      this.setState({account});
+      this.setState({companyName: this.state.account.company_name});
+    } else {
```

Comment:
we can get rid of this block since an account should always exist :) 

(the only reason we had to do this in the profile is because of some tech debt where some users don't have a profile set up yet, so if the profile failed to save we needed to reset all the values)